src/site/content/en/blog/payment-and-address-form-best-practices/index.md

```diff
@@ -0,0 +1,782 @@
+
+Postal code address suggestion doesn't work for all countries—and in some regions, post codes can 
+include a large number of potential addresses. 
+
+<figure class="w-figure">
+   <video controls autoplay loop muted class="w-screenshot">
+     <source src="https://samdutton.com/long-list-of-addresses.mp4" type="video/mp4">
+   </video>
+  <figcaption class="w-figcaption">ZIP or postal codes may include a lot of addresses!</figcaption>
+</figure>
+
+It's difficult for users to select from a long list of addresses—especially on mobile if they're 
+rushed or stressed. It can be easier and less error prone to let users take advantage of autofill, 
+and enter their complete address filled with a single tap or click.
+
+<figure class="w-figure">
+   <video controls autoplay loop muted class="w-screenshot">
+     <source src="https://samdutton.com/full-name-autofill.mp4" type="video/mp4">
+   </video>
+  <figcaption class="w-figcaption">A single name input enables one-tap (one-click) address entry.</figcaption>
+</figure>
+
+* Use [appropriate payment card autocomplete values](#payment-form-autocomplete).
+* Use a [single input for payment card numbers](#single-number-input).
+* [Avoid using custom elements](#avoid-custom-elements) that break the autofill experience.
+* [Test in the field as well as the lab](#analytics-rum): page analytics, interaction analytics, and 
+real-user performance measurement.
+* [Test across browsers, devices and platforms](#test-platforms). 
+
+## Simplify payment forms {: #general-guidelines}
+
+Payment forms are the single most critical part of the checkout process. Poor payment form design is 
+a [common cause of shopping cart abandonment](https://www.comscore.com/Insights/Presentations-and-Whitepapers/2017/Mobiles-Hierarchy-of-Needs). The [devil's in the details](https://en.wikipedia.org/wiki/The_devil_is_in_the_detail#cite_note-Titelman-1): small 
+glitches can tip users towards abandoning a purchase, particularly on mobile. Your job is to design 
+forms to make it as easy as possible for users to enter data.
+
+### Help users avoid re-entering payment data {: #payment-form-autocomplete}
+
+Make sure to add appropriate `autocomplete` values in payment card forms, including the payment card 
+number, name on the card, and the expiry month and year:
+* `cc-number`
+* `cc-name`
+* `cc-exp-month`
+* `cc-exp-year`
+
+This enables browsers help users by securely storing payment card details and correctly entering form 
+data. Without autocomplete, users may be more likely to keep a physical record of payment card details, 
+or store payment card data insecurely on their device.
+
+{% Aside 'caution' %}
+Don't add a selector for payment card type, since this can always be inferred from the payment card 
+number.
+{% endAside %}
+
+### Avoid using custom elements for payment card dates
+
+Custom elements can interrupt payment flow by breaking autofill and they don't work on older browsers. 
```

Comment:
Is this "The custom elements"? https://developer.mozilla.org/docs/Web/API/Window/customElements

Add a link?

Reply:
Added a link (and tweaked the text slightly): https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements

Reply:
BTW — this was a point commented on by Dominic in the script: 'Don't use custom <select> elements ... There is very little value but you break the autofill experience and even users who memorize their CVC need to walk to their credit card to look the expiration date up.'

Reply:
I was hoping this would be possible on a [`mwc-formfield`](https://github.com/material-components/material-components-web-components/issues/82). Haven't read closely yet but [for example](https://web.dev/more-capable-form-controls/#void-formresetcallback()).